### PR TITLE
add the role version number to the ansible-galaxy error/warning message

### LINKED ISSUE #52111: ansible-galaxy has cryptic error message when role version is wrong
##### SUMMARY
when specifying the wrong version for a ROLE that is downloading from GIT, ansible-galaxy fails with a cryptic message instead of complaining on missing tag/version 

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ansible-galaxy 

##### ANSIBLE VERSION
```
ansible-galaxy 2.7.6
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/srv/var/jenkins/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /srv/var/jenkins/.local/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible-galaxy
  python version = 2.7.5 (default, Jul 13 2018, 13:06:57) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]

```

##### CONFIGURATION
```
```

##### OS / ENVIRONMENT
Centos 7 

##### STEPS TO REPRODUCE

```
[jenkins@vm61 dummy]$ ansible-galaxy --version
ansible-galaxy 2.7.6
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/srv/var/jenkins/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /srv/var/jenkins/.local/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible-galaxy
  python version = 2.7.5 (default, Jul 13 2018, 13:06:57) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
[jenkins@vm61 dummy]$ cat roles.yml
# Standards: 0.1
---
- src: ssh://git.myserver.net:29418/operations/ansible/dummy
  scm: git
  version: '0.2'

[jenkins@vm61 dummy]$ ansible-galaxy install -r roles.yml  --force
 [WARNING]: - dummy was NOT installed successfully: - command /bin/git checkout 0.2 failed in directory /srv/var/jenkins/.ansible/tmp/ansible-local-22840YMo_1g/tmpi_69sw/dummy (rc=1)

ERROR! - you can use --ignore-errors to skip failed roles and finish processing the list.
[jenkins@vm61 dummy]$ 

```

##### EXPECTED RESULTS

ansible-galaxy provides error about missing tag/version 

##### ACTUAL RESULTS
the error bellow is not clear enough on why the checkout failed, can lead to hours of troubleshooting and code reading .... 
```
 [WARNING]: - dummy was NOT installed successfully: - command /bin/git checkout 0.2 failed in directory /srv/var/jenkins/.ansible/tmp/ansible-local-21277UaRVoO/tmp7UCvAY/dummy (rc=1)
```

### PR DESCRIPTION
**Pull request by jainnikhil30**
##### SUMMARY
cc @sudhirmohanraj, This is taken from your original PR #52853 which was closed.

Fixes: #52111 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

ansible-galaxy
##### ADDITIONAL INFORMATION

Before:
```
[niks@localhost]$ ansible-galaxy install -r requirements.yml 
- downloading role 'dummy', owned by 
[WARNING]: - dummy was NOT installed successfully: None (HTTP Code: 400, Message: Bad Request)
ERROR! - you can use --ignore-errors to skip failed roles and finish processing the list.
```

After this PR:
```
(venv) [niks@localhost ]$ ansible-galaxy install -r /home/niks/ansible_test/requirements.yml 
[WARNING]: You are running the development version of Ansible. You should only run Ansible from "devel" if you are modifying the Ansible engine, or trying out features under development. This is a rapidly changing source of code and can
become unstable at any point.
Starting galaxy role install process
- downloading role 'dummy', owned by 
[WARNING]: - dummy 1.1 was NOT installed successfully: None (HTTP Code: 400, Message: Bad Request)
ERROR! - you can use --ignore-errors to skip failed roles and finish processing the list.
```

### CODE REVIEW

File: lib/ansible/cli/galaxy.py [hunk 1]
```diff
@@ -1122,7 +1122,7 @@ def _execute_install_role(self, requirements):
             try:
                 installed = role.install()
             except AnsibleError as e:
-                display.warning(u"- %s was NOT installed successfully: %s " % (role.name, to_text(e)))
+                display.warning(u"- %s %s was NOT installed successfully: %s " % (role.name, role.version, to_text(e)))
```
Comment (samdoran) [inline lib/ansible/cli/galaxy.py]: Not all roles have a version. We should handle that by either not including `role.version` in the message or setting it to `*`.

Comment (jainnikhil30) [reply lib/ansible/cli/galaxy.py]: Sure. I have made that change. Thanks for the review @samdoran

**Comment by samdoran** [comment]
Looking at the original issue more closely, I don't think this fixes it. Simply displaying the version or `*` in the error message will just continue to mask the problem.

The real issue is the git checkout failure because a non-existing version was requested. That is what the error message should state.

**Comment by Akasurde** [comment]
@samdoran Could you please take a look? Thanks.

**Comment by samdoran** [comment]
It still seems like this change is not helping to better identify the real underlying cause.

**Comment by jainnikhil30** [comment]
@samdoran 

Thanks for your review. I will try and make that error more explicit in terms of version not being available remotely.

**Comment by Akasurde** [comment]
@jainnikhil30 Could you please address the @samdoran comment?

needs_info

**Pull request closed without merging**